pkg/server/users.go

```diff
@@ -609,3 +623,87 @@ func SlowDown() {
 	delay := time.Duration(rand.Int64N(2000)+2500) * time.Millisecond
 	time.Sleep(delay)
 }
+
+// Syncs user create/update events with each configured webhook endpoint using a
+// HTTP POST request with the created/modified [api.User] as the JSON body. This
+// reuses the access token in the [gin.Context] to authenticate the request to
+// the endpoint. It will log all errors, but will not handle the errors.
+func (s *Server) syncUserPost(c *gin.Context, user *api.User) {
+	for _, u := range s.conf.UserSync.WebhookURLs() {
+		var (
+			req       *http.Request
+			bodyBytes []byte
+			token     string
+			resp      *http.Response
+			err       error
+		)
+
+		// Marshal the user into JSON bytes
+		if bodyBytes, err = json.Marshal(user); err != nil {
+			log.Warn().Err(err).Str("endpoint_url", u.String()).Str("user_id", user.ID.String()).Msg("user sync post: could not marshal user to json")
+			return
+		}
+
+		// Create a POST request for JSON
+		if req, err = http.NewRequestWithContext(c.Request.Context(), http.MethodPost, u.String(), bytes.NewReader(bodyBytes)); err != nil {
+			log.Warn().Err(err).Str("endpoint_url", u.String()).Str("user_id", user.ID.String()).Msg("user sync post: could not create new post request")
+			return
+		}
+		req.Header.Set("Content-Type", "application/json")
+
+		// Add authorization token
+		if token, err = gimauth.GetAccessToken(c); err != nil || token == "" {
+			log.Warn().Err(err).Str("endpoint_url", u.String()).Str("user_id", user.ID.String()).Msg("user sync post: could not attain an access token from context")
+			return
+		}
+		req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", token))
+
+		// Do request
+		if resp, err = http.DefaultClient.Do(req); err != nil {
+			log.Warn().Err(err).Str("endpoint_url", u.String()).Str("user_id", user.ID.String()).Str("response_status", resp.Status).Msg("user sync post: could not complete http post request")
+			return
+		}
```

Comment:
`resp.Body` will need to be closed since the client expects the user to handle [this](https://pkg.go.dev/net/http#Client.Do). The change will also have to be applied in `syncUserDelete`.